Cargo.toml

```diff
@@ -27,18 +28,34 @@ members = [
 ]
 
 [workspace.dependencies]
+# The order matches with publishing order.
+sails-macros-core = { path = "sails/macros/core" }
+sails-macros = { path = "sails/macros" }
+sails-idl-parser = { path = "sails/idl-parser" }
+sails-client-gen = { path = "sails/client-gen" }
+sails = { package = "sails-rs", path = "sails" }
+sails-idl-gen = { path = "sails/idl-gen" }
+
+# Gear deps
+gclient = "=1.4.2"
+gear-core-errors = "=1.4.2"
+gprimitives = { version = "=1.4.2", features = ["codec"] }
+gstd = "=1.4.2"
+gtest = "=1.4.2"
+gwasm-builder = { package = "gear-wasm-builder", version = "=1.4.2" }
+
+# Shared with gear deps
+parity-scale-codec = { version = "3.6", default-features = false }
+primitive-types = { version = "0.12", default-features = false }
+scale-info = { version = "2.11", default-features = false }

```

Comment:
I'm not quite convinced with the changed order - it is difficult to decide where to put/find a dependency. You say `Shared with gear` - I'm pretty sure there are plenty of other deps which are shared with gear, but you highlight just some of them due to some reason. Alphabetical order is much more convinient and easy to use

Reply:
Gear deps are the most important ones. I hard pinned it to 1.4.2 version, so it's better to keep somewhere together. This section is for gear protocol libraries only (you can mean from "gear-tech/gear" source.

Speaking of "shared" you're totally right. My motivation was to pin the same codec and scale info version as in gear libs to avoid hypothetical collisions but decided to deal with it some time soon (meanwhile cargo kindly manages versions around gear's pinned, so its okay). If you want to remove this section feel free to do it after merging this in the most comfortable for you way. But I'd insist of separation local crates and gear deps as dedicated sections